rust/src/command/internal/mod.rs

```diff
@@ -76,74 +76,53 @@ impl InternalCommand {
-                    };
+                AccountDiff::FeeTransfer(this_account_diff_fee) => {
+                    assert!(
+                        n + 1 < account_diff_fees.len(),
+                        "Missing a debit/credit pair"
+                    );
 
-                    // aggregate
-                    if let Some((idx, cmd)) =
-                        internal_cmds
-                            .iter()
-                            .enumerate()
-                            .find_map(|(m, cmd)| match cmd {
-                                Self::FeeTransfer {
-                                    sender,
-                                    receiver,
-                                    amount,
-                                } => {
-                                    if *sender == ic_sender && *receiver == ic_receiver {
-                                        return Some((
-                                            m,
-                                            Self::FeeTransfer {
-                                                sender: ic_sender.clone(),
-                                                receiver: ic_receiver.clone(),
-                                                amount: f.amount.0 + *amount,
-                                            },
-                                        ));
-                                    }
-                                    None
-                                }
-                                _ => None,
-                            })
-                    {
-                        internal_cmds.push(cmd);
-                        internal_cmds.swap_remove(idx);
-                    } else {
-                        internal_cmds.push(Self::FeeTransfer {
-                            sender: ic_sender.clone(),
-                            receiver: ic_receiver.clone(),
-                            amount: f.amount.0,
-                        });
-                    }
+                    let next_account_diff_fee = &account_diff_fees[n + 1];
+                    let (ic_sender, ic_receiver) =
+                        if this_account_diff_fee.update_type == UpdateType::Credit {
+                            (
+                                next_account_diff_fee.public_key(),
+                                this_account_diff_fee.public_key.clone(),
+                            )
+                        } else {
+                            (
+                                this_account_diff_fee.public_key.clone(),
+                                next_account_diff_fee.public_key(),
+                            )
+                        };
+
+                    internal_cmds.push(Self::FeeTransfer {
+                        sender: ic_sender.clone(),
+                        receiver: ic_receiver.clone(),
+                        amount: this_account_diff_fee.amount.0
+                            + next_account_diff_fee.amount().try_into().ok().unwrap_or(0),
```

Comment:
This is highly suspect. When dealing with token amounts we should never modify the intended value. Modifying the value is a bug. If we can't convert, throw a noisy Error